sycl/test-e2e/Matrix/joint_matrix_gemm_cuda.hpp

```diff
@@ -168,20 +168,15 @@ void test(queue &q) {
 
               // round values to correct precision if using tf32
               if constexpr (std::is_same<T3, precision::tf32>::value) {
-                auto wi_size = get_wi_data(sg, sub_a).length();
-                assert(wi_size == get_wi_data(sg, sub_b).length());
-                for (auto i = 0; i < wi_size; ++i) {
-                  get_wi_data(sg, sub_a)[i] =
-                      round_to_tf32(get_wi_data(sg, sub_a)[i]);
-                  get_wi_data(sg, sub_b)[i] =
-                      round_to_tf32(get_wi_data(sg, sub_b)[i]);
-                }
+                auto round_lambda = [](auto &x) { x = round_to_tf32(x); };
+                joint_matrix_apply(sg, sub_a, round_lambda);
+                joint_matrix_apply(sg, sub_b, round_lambda);
               }
 
-              joint_matrix_mad(sg, sub_c, sub_a, sub_b, sub_c);
+              joint_matrix_mad(sg, sub_d, sub_a, sub_b, sub_c);
```

Comment:
how many times this will actually run?
You want the return matrix to be "sub_c" so it can be input again, right?

Reply:
Just once, I changed it to run once so that sub_d can be a different joint_matrix (with a different type to sub_c), in order to test the new cases I added to the support matrix/joint_matrix_tensorcores_sm70.cpp.